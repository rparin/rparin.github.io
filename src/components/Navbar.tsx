export default function Navbar() {
  return (
    <nav className="">
      <ul className="flex flex-row gap-10 md:gap-12 justify-center">
        <li>
          <a className="text-xl" href="#projects">
            Projects
          </a>
        </li>
        <li>
          <a className="text-xl" href="#about">
            About
          </a>
        </li>
        <li>
          <a className="text-xl" href="#contact">
            Contact
          </a>
        </li>
        <li>
          <a
            className="text-xl"
            href="https://raw.githubusercontent.com/rparin/rparin.github.io/main/public/Resume.pdf"
            target="_blank"
            rel="noopener noreferrer">
            Resume
          </a>
        </li>
      </ul>
    </nav>
  );
}
