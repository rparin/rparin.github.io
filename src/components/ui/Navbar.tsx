import { NavItems } from "@/constants/NavLinks";

export default function Navbar() {
  const NavTags = [];
  for (const [key, value] of Object.entries(NavItems)) {
    NavTags.push(
      <li key={value.id}>
        <a
          className="text-xl"
          href={value?.url ? value?.url : "#" + value.id}
          target={value?.url ? "_blank" : undefined}
          rel={value?.url ? "noopener noreferrer" : undefined}>
          {value.title}
        </a>
      </li>
    );
  }

  return (
    <>
      <nav className="sticky top-0 bg-background drop-shadow-lg z-10 ">
        <ul className="flex flex-row gap-9 md:gap-12 md:justify-center py-5 ml-5 pr-5 overflow-y-auto">
          {NavTags}
        </ul>
      </nav>
    </>
  );
}
