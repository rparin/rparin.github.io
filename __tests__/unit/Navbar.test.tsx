import { act, render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Navbar from "@/components/ui/Navbar";

describe("testing Navbar Component Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(<Navbar />);
    const results = await act(async () => axe(container));
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<Navbar />);
  expect(container).toMatchSnapshot();
});
