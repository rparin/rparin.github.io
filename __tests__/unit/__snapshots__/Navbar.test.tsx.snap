// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <nav
    class="sticky top-[-1px] z-10 bg-background drop-shadow-lg"
  >
    <ul
      class="ml-5 flex flex-row justify-center gap-7 overflow-y-auto py-3 pr-5 md:gap-10 md:py-4"
    >
      <li
        class="mt-1"
      >
        <a
          class=""
          href="#projects"
        >
          Projects
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class=""
          href="#about"
        >
          About
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class=""
          href="#contact"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          class="m-0 flex flex-row rounded-md bg-primary p-0 text-text hover:bg-secondary dark:bg-primary dark:text-text dark:hover:bg-secondary"
          href="/Ralph_Parin_Resume.pdf"
        >
          <span
            class="p-0 pb-1 pl-2 pr-2 pt-1"
          >
            Resume
          </span>
          <div
            class="flex justify-end rounded-r-md bg-secondary"
          >
            <svg
              class="h-auto w-8 p-2 invert dark:invert-0 sm:p-1.5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
              />
              <polyline
                points="15 3 21 3 21 9"
              />
              <line
                x1="10"
                x2="21"
                y1="14"
                y2="3"
              />
            </svg>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
